### PR TITLE
Add WithTestWsClient trait and fix HealthCheckTest

### PR DESCRIPTION
**Pull request by TBonnin**
## What does this change?
- Add WithTestWsClient trait so tests can access an instance of `WSClient` rather than calling `WS.url`
## What is the value of this and can you measure success?

=> Play 2.5
Cleaning more call to `WS.url()`
## Does this affect other platforms - Amp, Apps, etc?

NO
## Request for comment

@alexduf

### CODE REVIEW

File: common/test/package.scala [hunk 1]
```diff
@@ -59,7 +60,11 @@ trait TestSettings {
   ContentApiClient.jsonClient._http = toRecorderHttp(ContentApiClient.jsonClient._http)
 }
 
-trait ConfiguredTestSuite extends ConfiguredServer with ConfiguredBrowser with ExecutionContexts {
+trait WithTestWsClient {
+  lazy val wsClient = NingWSClient(NingWSClientConfig(maxRequestRetry = 0))
```
Comment (alexduf) [inline common/test/package.scala]: Two things here

- NingWsClient will need to be replaced by AhcWsClient when bumping to play 2.5. There's no way around it.
- When creating a wsClient yourself, you need to handle its destruction.

Here's an extract of the docs

> If you create a WSClient manually then you must call client.close() to clean it up when you’ve finished with it.

you can find it here https://www.playframework.com/documentation/2.5.x/ScalaWS#Using-WSClient

**Comment by TBonnin** [comment]
@alexduf made the change to close the wsClient. Feedback welcome :)

File: common/app/common/Lazy.scala [hunk 2]
```diff
@@ -0,0 +1,20 @@
+package common
+
+import scala.language.implicitConversions
+
+object Lazy {
+  def lazily[A](f: => A): Lazy[A] = new Lazy(f)
```
Comment (alexduf) [inline common/app/common/Lazy.scala]: I would rename this one `apply`

File: common/app/common/Lazy.scala [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  implicit def evalLazy[A](l: Lazy[A]): A = l()
+}
+
+class Lazy[A] private(f: => A) {
+  private var option: Option[A] = None
+
+  def apply(): A = option match {
```
Comment (alexduf) [inline common/app/common/Lazy.scala]: but this one `.get` :)

**Comment by alexduf** [comment]
The two comments I left are really minor, please feel free to merge as it looks good! Good idea to wrap this in a type :+1:

File: common/app/common/Lazy.scala (hunk 2 above)
Comment (TBonnin) [inline common/app/common/Lazy.scala]: Make sense

File: common/app/common/Lazy.scala (hunk 3 above)
Comment (TBonnin) [inline common/app/common/Lazy.scala]: Good point

**Comment by alexduf** [comment]
:star: :+1: :star:

**Pull request merged**